code/datums/recipe.dm

```diff
@@ -109,19 +117,104 @@ var/global/list/microwave_accepts_items =    list(
 	if(!result)
 		log_error("<span class='danger'>Recipe [type] is defined without a result, please bug this.</span>")
 		return
-	var/obj/result_obj = new result(container)
-	container.reagents?.clear_reagents()
-	//Checked here in case LAZYCLEARLIST nulls and no more physical ingredients are added
+	/*
+	We will subtract all the ingredients from the container, and transfer their reagents into a holder
+	We will not touch things which are not required for this recipe. They will be left behind for the caller
+	to decide what to do. They may be used again to make another recipe or discarded, or merged into the results,
+	thats no longer the concern of this proc
+	*/
+	var/datum/reagents/buffer = new /datum/reagents(1e12, global.temp_reagents_holder)//
 	var/list/container_contents = container.get_contained_external_atoms()
-	if(!container_contents)
-		return result_obj
-	for (var/obj/O in (container_contents-result_obj))
-		if (O.reagents)
-			O.reagents.clear_reagent(/decl/material/liquid/nutriment)
-			O.reagents.update_total()
-			O.reagents.trans_to_obj(result_obj, O.reagents.total_volume)
-		if(istype(O,/obj/item/holder/))
-			var/obj/item/holder/H = O
-			H.destroy_all()
-		qdel(O)
-	return result_obj
+	//Find items we need
+	if (LAZYLEN(items))
+		for (var/i in items)
+			var/obj/item/I = locate(i) in container_contents
+			if (I && I.reagents)
+				I.reagents.trans_to_holder(buffer,I.reagents.total_volume)
+				qdel(I)
+
+	//Find fruits
+	if (LAZYLEN(fruit))
+		var/list/checklist = list()
+		checklist = fruit.Copy()
+
+		for(var/obj/item/chems/food/snacks/grown/G in container_contents)
+			if(!G.seed || !G.seed.kitchen_tag || isnull(checklist[G.seed.kitchen_tag]))
+				continue
+
+			if (checklist[G.seed.kitchen_tag] > 0)
+				//We found a thing we need
+				checklist[G.seed.kitchen_tag]--
+				if (G && G.reagents)
+					G.reagents.trans_to_holder(buffer,G.reagents.total_volume)
+				qdel(G)
+
+	//And lastly deduct necessary quantities of reagents
+	if (LAZYLEN(reagents))
+		for (var/r in reagents)
+			//Doesnt matter whether or not there's enough, we assume that check is done before
+			container.reagents.trans_type_to_holder(buffer, r, reagents[r])
+
+	/*
+	Now we've removed all the ingredients that were used and we have the buffer containing the total of
+	all their reagents.
+	If we have multiple results, holder will be used as a buffer to hold reagents for the result objects.
+	If, as in the most common case, there is only a single result, then it will just be a reference to
+	the single-result's reagents
+	*/
+	var/datum/reagents/holder = new/datum/reagents(1e12, global.temp_reagents_holder)
```

Comment:
1e12?

Reply:
Large amount.

Reply:
`INFINITY` would be clearer.

Reply:
Can you actually make a container with infinite reagents? I mean, I guess you *can...*

Reply:
It's as valid as making one with `1e12` reagents for all practical purposes.